tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -546,21 +593,25 @@ def check_model_object(argv):
             else:
                 raise Error("Unknown checkpoint format.")
 
-        if isinstance(model, tf.keras.layers.Layer) or isinstance(model, tf.Module):
-            assert 'input_shape' in argv and argv['input_shape'] is not None, \
-                "Converting of {} requires providing of input_shape.".format(type(model))
-            assert len(argv['input_shape']) > 0, "Please provide non-empty input shape."
-            inputs = []
-            for shape_idx, shape in enumerate(parse_input_shapes(argv)):
-                inp_shape = get_static_shape(shape)
-                batch_size = None
-                if len(inp_shape) > 1:
-                    batch_size = inp_shape[0]
-                    inp_shape = inp_shape[1:]
-                inputs.append(tf.keras.Input(shape=inp_shape, batch_size=batch_size))
-            outputs = model(*inputs)
-            argv['input_model'] = tf.keras.Model(inputs, outputs)
-            argv['input_shape'] = None
+        if isinstance(model, (tf.keras.layers.Layer, tf.Module, tf.keras.Model)):
+            return "tf"
+        if isinstance(model, Trackable):
+            if hasattr(model, 'signatures') and len(model.signatures.items()):
+                if 'serving_default' in model.signatures:
+                    argv['input_model'] = model.signatures['serving_default']
+                elif 'default' in model.signatures:
+                    argv['input_model'] = model.signatures['default']
+                else:
+                    for signature_name, signature in model.signatures.items():
+                        argv['input_model'] = model.signatures[signature_name]
+                        log.warning("Could not find the default signature. "
+                                    "The following signature was used for conversion: {}".format(signature_name))
+                        break
+
+            elif hasattr(model, 'graph'):
+                argv['input_model'] = model.graph
+            else:
+                raise Error("Could not find signature of graph in a Trackable object.")
```

Comment:
I believe that there should be some Python TF FE routine that will check if some memory object is supported by TF FE. And move it here. This is a sort of `load_by_model` method of Frontend manager that return needed frontend by the model format.

@slyalin

Reply:
Probably Python Frontend Manager should contact each frontend and ask if it supports this Python memory object.
Moreover, there should be universal point for both models from memory and disk.
All kind of memory and disk specific things should be resolved on frontend managed and frontend layers, not on MO side at all.